akka-docs/rst/scala/code/docs/persistence/PersistenceEventAdapterDocSpec.scala

```diff
@@ -0,0 +1,233 @@
+    event // identity
+
+  override def fromJournal(event: Any, manifest: String): EventSeq =
+    EventSeq.single(event) // identity
+}
+//#identity-event-adapter
+
+/**
+ * This is an example adapter which completely takes care of domain<->json translation.
+ * It allows the journal to take care of the manifest handling, which is the FQN of the serialized class.
+ */
+class MyAutoJsonEventAdapter(system: ExtendedActorSystem) extends EventAdapter {
+  private val gson = new Gson
+
+  override def manifest(event: Any): String =
+    event.getClass.getCanonicalName
+
+  override def toJournal(event: Any): Any = gson.toJsonTree(event)
+
+  override def fromJournal(event: Any, manifest: String): EventSeq = EventSeq.single {
+    event match {
+      case json: JsonElement =>
+        val clazz = system.dynamicAccess.getClassFor[Any](manifest).get
+        gson.fromJson(json, clazz)
+    }
+  }
+}
+
+class MyUpcastingEventAdapter(system: ExtendedActorSystem) extends EventAdapter {
+  override def manifest(event: Any): String = ""
+
+  override def toJournal(event: Any): Any = ???
+
+  override def fromJournal(event: Any, manifest: String): EventSeq = ???
+}
+
+/**
+ * This is an example adapter which completely takes care of domain<->json translation.
+ * It manually takes care of smuggling through the manifest even if the journal does not do anything in this regard,
+ * which is the case in this example since we're persisting to the inmem journal, which does nothing in terms of manifest.
+ */
+class MyManualJsonEventAdapter(system: ExtendedActorSystem) extends EventAdapter {
+
+  private val gson = new Gson
+
+  override def manifest(event: Any): String = event.getClass.getCanonicalName
+
+  override def toJournal(event: Any): Any = {
+    val out = gson.toJsonTree(event).getAsJsonObject
+
+    // optionally can include manifest in the adapted event.
+    // some journals will store the manifest transparently
+    out.addProperty("_manifest", manifest(event))
+
+    out
+  }
+
+  override def fromJournal(event: Any, m: String): EventSeq = event match {
+    case json: JsonElement =>
+      val manifest = json.getAsJsonObject.get("_manifest").getAsString
+
+      val clazz = system.dynamicAccess.getClassFor[Any](manifest).get
+      EventSeq.single(gson.fromJson(json, clazz))
+  }
+}
+
+class MyTaggingEventAdapter(system: ExtendedActorSystem) extends EventAdapter {
+  override def manifest(event: Any): String = ""
+
+  override def fromJournal(event: Any, manifest: String): EventSeq = event match {
+    case j: MyTaggingJournalModel => EventSeq.single(j)
+  }
+
+  override def toJournal(event: Any): Any = {
+    event match {
+      case Person(_, age) if age >= 18 => MyTaggingJournalModel(event, tags = Set("adult"))
+      case Person(_, age)              => MyTaggingJournalModel(event, tags = Set("minor"))
+      case _                           => MyTaggingJournalModel(event, tags = Set.empty)
+    }
+  }
+}
+
```

Comment:
Docs here are incomplete (impls not included in docs, but that will be part of the documenting schema evolution ticket)